File: 백준/Silver/33754. 격자 막기/격자 막기.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args)
    {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int[] a = new int[n];

        int min = 2;

        for (int i = 0; i < n; i++)
        {
            a[i] = input.nextInt();
            if (a[i] == 0)
                min = 1;
        }

        for (int i = 0; i < n; i++)
        {
            if (input.nextInt() == 0)
            {
                if (a[i]==0 || i>0 && a[i-1]==0 || a[i+1]==0 && i<n-1)
                {
                    min = 0;
                }
            }
        }

        System.out.println(min);
    }
}

File: 백준/Silver/33754. 격자 막기/격자 막기.py
N = int(input())

a = list(map(int, input().split()))
b = list(map(int, input().split()))

ans = 2

for i in range(1, N):
    if a[i] == 0:
        if b[i] == 0 or b[i-1] == 0 or (i+1<N and b[i+1] == 0):
            ans = min(ans, 0)
        else:
            ans = min(ans, 1)
    if b[i-1] == 0:
        ans = min(ans, 1)
print(ans)